name: Django-app workflow

on:
    push:
      branches-ignore:
        - master
jobs:

  tests:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: 3.7

    - name: Install dependencies
      run: | 
        python -m pip install --upgrade pip 
        pip install flake8 pep8-naming flake8-broken-line flake8-return
        pip install -r api_yamdb/requirements.txt 

    - name: Test with flake8
      run: |
        python -m flake8

    - name: Django tests
      run: |
        pytest

    - name: Send message if Tests failed
      if: ${{ failure() }}
      uses: appleboy/telegram-action@master
      with:
        to: ${{ secrets.TELEGRAM_TO }}
        token: ${{ secrets.TELEGRAM_TOKEN }}
        message: |
          Ошибка при тестировании - ${{ github.repository }}
          https://github.com/${{ github.repository }}actions/runs/${{ github.run_id }}
